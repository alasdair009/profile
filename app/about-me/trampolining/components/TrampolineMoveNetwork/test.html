<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
