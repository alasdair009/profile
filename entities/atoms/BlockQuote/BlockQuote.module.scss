@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/typography";
@use "../../../app/styles/helpers/rem";
@use "sass:map";

.root {
  display: block;
  font-size: rem.toRem(map.get(typography.$fontSizes, large));
  font-style: italic;
  margin: rem.toRem(map.get(dimensions.$sizes, s32)) auto;
  max-width: rem.toRem(dimensions.$globalTextMaxWidth);
  padding: rem.toRem(map.get(dimensions.$sizes, s8));
  position: relative;
  text-align: center;
  text-wrap: balance;

  &::before,
  &::after {
    color: colors.$colorGreenGrass;
    content: '"';
    font-size: rem.toRem(map.get(typography.$fontSizes, xxlarge));
    font-weight: map.get(typography.$fontWeights, black);
  }
}
