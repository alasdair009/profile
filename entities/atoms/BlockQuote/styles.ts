"use client";
import styled from "styled-components";
import {
  colors,
  fontSizes,
  fontWeights,
  globalTextMaxWidth,
  sizes,
} from "@/entities";

export const Root = styled.blockquote`
  display: block;
  font-size: ${fontSizes.large.rem};
  font-style: italic;
  margin: ${sizes.s32.rem} auto;
  max-width: ${globalTextMaxWidth};
  padding: ${sizes.s8.rem};
  position: relative;
  text-align: center;
  text-wrap: balance;

  &::before,
  &::after {
    color: ${colors.greenGrass};
    content: '"';
    font-size: ${fontSizes.xxlarge.rem};
    font-weight: ${fontWeights.black};
  }
`;
