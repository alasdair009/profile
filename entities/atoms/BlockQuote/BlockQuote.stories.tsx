import { BlockQuote } from "./BlockQuote";
import { Meta, StoryObj } from "@storybook/react";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof BlockQuote> = {
  component: BlockQuote,
  args: {
    children: "Some cool quote",
  },
};
export default meta;

export const Default: StoryObj<typeof BlockQuote> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const blockQuoteElement = canvas.getByTestId(BlockQuote.name);

    await expect(blockQuoteElement).toBeInTheDocument();
    await expect(blockQuoteElement).toHaveTextContent(`${args.children}`);
  },
};
