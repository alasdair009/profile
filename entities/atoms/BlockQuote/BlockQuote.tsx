import { Root } from "./styles";
import { HTMLAttributes } from "react";

/**
 * A large highlight of a short string.
 */
export function BlockQuote({ ...rest }: HTMLAttributes<HTMLSpanElement>) {
  return <Root {...rest} />;
}
