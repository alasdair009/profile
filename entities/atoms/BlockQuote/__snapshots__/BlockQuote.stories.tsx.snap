// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/BlockQuote Default play-test 1`] = `
<blockquote data-testid="BlockQuote"
            class="sc-hDcvty qmgA-D"
>
  Some cool quote
</blockquote>
`;
